### PR TITLE
Use lazy import for Matplotlib and move cache to runpath

### PR DESCRIPTION
**Pull request by zsambokiandras**
## Bug / Requirement Description
When one runs Testplan scripts that include matplotlib you get a warning about matplotlib not being able to write to a disted path.

`Matplotlib created a temporary config/cache directory at /tmp/matplotlib-xxxxxxxx because the default path (xxx/.config/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.`

## Solution description
Set matplotlib default config/cache path. We can use MPLCONFIGDIR env var to define where matplotlib can write. Use testplan runtime path.

## Checklist:
- [x] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [x] News fragment present for release notes
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/runnable/base.py [hunk 1]
```diff
@@ -1054,12 +1054,21 @@ def make_runpath_dirs(self):
...
         if self.cfg.path_cleanup is False:
             makedirs(self._runpath)
             makedirs(self._scratch)
+            if not env_mpl_cache:
+                makedirs(self._mplcache)
         else:
             makeemptydirs(self._runpath)
             makeemptydirs(self._scratch)
+            if not env_mpl_cache:
+                makeemptydirs(self._mplcache)
```
Comment (Pyifan) [inline testplan/runnable/base.py:1071]: i don't think we shall always create this dir for matplotlib, 99% of time we don't need this dir. Now that the warning msg is only issued when matplotlib is actually used by pdf exporter, it's OK to leave it to user to set it.

**Review by Pyifan — CHANGES_REQUESTED** [review]

File: testplan/runnable/base.py (hunk 1 above)
Comment (zsambokiandras) [reply testplan/runnable/base.py]: Removed the creation, now matplotlib automatically creates the dir. Should I also remove setting the env var to the runpath, if not set by the user?

File: testplan/runnable/base.py [hunk 2]
```diff
@@ -1072,6 +1072,11 @@ def make_runpath_dirs(self):
             )
             makedirs(self.resource_monitor_server_file_path)
 
+        if not os.environ.get("MPLCONFIGDIR"):
+            os.environ["MPLCONFIGDIR"] = os.path.join(
+                self._runpath, "matplotlib"
```
Comment (Pyifan) [inline testplan/runnable/base.py:1077]: does matplotlib complain if the dir pointed by MPLCONFIGDIR doesn't exist?

Comment (zsambokiandras) [reply testplan/runnable/base.py:1077]: No, it tries to create that.

This sets the path to the runpath, if the user didn't set the env var. This way there will be no warning, and the dir is only created when the matplotlib import executes. If I remove this default setting, there will be a warning when matplotlib is imported (and the default location is read-only).

File: testplan/exporters/testing/pdf/renderers/entries/baseUtils.py [hunk 3]
```diff
@@ -81,6 +77,11 @@ def get_matlib_plot(source):
     Call the appropriate plotting function based on
     whether a graph or chart is being plotted.
     """
+    import matplotlib
+
+    matplotlib.use("Agg")
+    import matplotlib.pyplot as plot
```
Comment (zhenyu-ms) [inline testplan/exporters/testing/pdf/renderers/entries/baseUtils.py:83]: maybe this line can be removed?

**Review by Pyifan — APPROVED** [review]

**Review by Pyifan — APPROVED** [review]

**Pull request merged**